### PR TITLE
Merges ANCM into IISIntegration

### PR DESCRIPTION
**Pull request by jkotalik**
@pan-wang there are a few build warnings in IISLib, may I fix those before merging?

There is a PR open in [ANCM ](https://github.com/aspnet/AspNetCoreModule/pull/274) that actually creates the RequestHandler nuget package. Depending on if that gets approved, I will add it on here.

@natemcmaster I mostly need you to check build stuff.

cc/ @Eilon @muratg as this starts the process of deprecating the ANCM repo.

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -12,3 +12,27 @@ Unless required by applicable law or agreed to in writing, software distributed
 under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
```
Comment (jkotalik) [inline LICENSE.txt]: @Eilon I just appended the ANCM LICENSE file to IISIntegration's, would you prefer I mention "IISIntegration uses two licenses files..."

File: README.md [hunk 2]
```diff
@@ -1,6 +1,80 @@
-ASP.NET Core IIS Integration
-========
-
-This repo hosts the ASP.NET Core middleware for IIS integration.
+This repo hosts the ASP.NET Core middleware for IIS integration and the AspNetCoreModule
 
 This project is part of ASP.NET Core. You can find samples, documentation and getting started instructions for ASP.NET Core at the [Home](https://github.com/aspnet/home) repo.
+
```
Comment (jkotalik) [inline README.md]: @shirhatti I think I should just remove this section of the README and we can update it later.

File: build/dependencies.props [hunk 3]
```diff
@@ -3,41 +3,46 @@
...
-    <SystemNumericsVectorsPackageVersion>4.5.0-preview1-25902-08</SystemNumericsVectorsPackageVersion>
-    <SystemRuntimeCompilerServicesUnsafePackageVersion>4.5.0-preview1-25902-08</SystemRuntimeCompilerServicesUnsafePackageVersion>
-    <SystemSecurityPrincipalWindowsPackageVersion>4.4.0</SystemSecurityPrincipalWindowsPackageVersion>
-    <SystemTextEncodingsWebUtf8PackageVersion>0.1.0-alpha-002</SystemTextEncodingsWebUtf8PackageVersion>
-    <XunitPackageVersion>2.3.0</XunitPackageVersion>
-    <XunitRunnerVisualStudioPackageVersion>2.3.0</XunitRunnerVisualStudioPackageVersion>
+    <MicrosoftWebAdministrationPackageVersion>7.0.0</MicrosoftWebAdministrationPackageVersion>
+    <SystemBuffersPackageVersion>4.5.0-preview1-26016-05</SystemBuffersPackageVersion>
+    <SystemIOPipelinesPackageVersion>0.1.0-e171215-1</SystemIOPipelinesPackageVersion>
+    <SystemManagementAutomationDllPackageVersion>10.0.10586</SystemManagementAutomationDllPackageVersion>
```
Comment (jkotalik) [inline build/dependencies.props]: @natemcmaster There was a dll dependency in AspNetCoreModule tests. Would this need to be an external dependency in universe or is there another way to remove this?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,80 @@
-ASP.NET Core IIS Integration
-========
-
-This repo hosts the ASP.NET Core middleware for IIS integration.
+This repo hosts the ASP.NET Core middleware for IIS integration and the AspNetCoreModule
```
Comment (natemcmaster) [inline README.md]: Nit: "ASP.NET Core Module"

File: README.md [hunk 5]
```diff
@@ -1,6 +1,80 @@
...
 
 This project is part of ASP.NET Core. You can find samples, documentation and getting started instructions for ASP.NET Core at the [Home](https://github.com/aspnet/home) repo.
+
+=======
+# ASP.NET Core Module
+
+The ASP.NET Core Module is an IIS Module which is responsible for process
+management of ASP.NET Core http listeners and to proxy requests to the process
```
Comment (natemcmaster) [inline README.md]: Nit: casing HTTP

File: README.md [hunk 6]
```diff
@@ -1,6 +1,80 @@
...
+that it manages.
+
+## Installing the latest ASP.NET Core Module
+The ASP.NET Core Module for IIS can be installed on servers without installing the .NET Core runtime. You can download the [Windows (Server Hosting) installer](https://go.microsoft.com/fwlink/?linkid=832756) and run the following command from an Administrator command prompt:
+``DotNetCore.1.1.0.Preview1-WindowsHosting.exe OPT_INSTALL_LTS_REDIST=0 OPT_INSTALL_FTS_REDIST=0``
```
Comment (natemcmaster) [inline README.md]: This is outdated. Should we just remove it?

File: README.md [hunk 7]
```diff
@@ -1,6 +1,80 @@
...
+
+## Pre-requisites for building
+
+### Windows 8.1+ or Windows Server 2012 R2+
+
+### Visual C++ Build Tools
+
+[Download](http://download.microsoft.com/download/D/2/3/D23F4D0F-BA2D-4600-8725-6CCECEA05196/vs_community_ENU.exe)
```
Comment (natemcmaster) [inline README.md]: Let's update this to VS2017.

File: build/dependencies.props (hunk 3 above)
Comment (natemcmaster) [reply build/dependencies.props]: Is it this package? https://www.nuget.org/packages/System.Management.Automation.dll/

I'm not sure what this assembly is used for. Also, not sure who "bigfoot" is, but it looks like he took a PowerShell assembly and wrapped it up in a nupkg.

File: build/repo.targets [hunk 8]
```diff
@@ -0,0 +1,45 @@
+<Project>
+  <PropertyGroup>
+    <PublishFeed>https://dotnet.myget.org/F/aspnetcoremodule/api/v2/package</PublishFeed>
```
Comment (natemcmaster) [inline build/repo.targets]: You can remove this now. We don't need this feed.

File: build/repo.targets [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      <PackageVersion>1.0.0-pre-$(BuildNumber)</PackageVersion>
+    </PropertyGroup>
+
+    <MSBuild
+      Projects="$(MSBuildThisFileDirectory)../nuget/AspNetCore.csproj"
+      Targets="Restore;Pack"
+      Properties="PackageVersion=$(PackageVersion);PackageOutputPath=$(BuildDir)" />
+  </Target>
+
+  <Target Name="PublishPackage" Condition="'$(PublishPackage)'=='true'">
```
Comment (natemcmaster) [inline build/repo.targets]: You can remove this target altogether. Universe will take care of publishing.

File: build/repo.targets [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      
+    <Exec Command="&quot;$(VisualStudioMSBuildx86Path)&quot; &quot;$(RepositoryRoot)src\AspNetCore\AspNetCore.vcxproj&quot; %(BuildConfigurations.Identity)"
+      Condition="'$(VisualStudioMSBuildx86Path)' != ''" />
+    <Exec Command="&quot;$(VisualStudioMSBuildx86Path)&quot; &quot;$(RepositoryRoot)src\RequestHandler\RequestHandler.vcxproj&quot; %(BuildConfigurations.Identity)"
+      Condition="'$(VisualStudioMSBuildx86Path)' != ''" />
+  </Target>
+
+  <Target Name="PackageProjects">
+    <PropertyGroup>
+      <PackageVersion>1.0.0-pre-$(BuildNumber)</PackageVersion>
```
Comment (natemcmaster) [inline build/repo.targets]: You can remove this too. It is already be set in version.props.

https://github.com/aspnet/IISIntegration/blob/d2fba28272759f9c59ca9e08fd9d4373576a919d/version.props#L5-L6

File: build/repo.props [hunk 11]
```diff
@@ -1,12 +1,20 @@
-﻿<Project>
+<Project>
   <Import Project="dependencies.props" />
 
   <ItemGroup>
     <ExcludeFromTest Include="$(RepositoryRoot)test\TestSites\*.csproj" />
     <ExcludeFromTest Include="$(RepositoryRoot)test\IISTestSite\*.csproj" />
     <ExcludeFromTest Include="$(RepositoryRoot)test\IISIntegration.FunctionalTests\*.csproj" Condition="'$(OS)' != 'Windows_NT'" />
     <ExcludeFromTest Include="$(RepositoryRoot)test\IISIntegration.IISServerFunctionalTests\*.csproj" Condition="'$(OS)' != 'Windows_NT'" />
+        <ExcludeFromTest Include="$(RepositoryRoot)test\AspNetCoreModule.TestSites.Standard\*.csproj" />
```
Comment (natemcmaster) [inline build/repo.props]: Nit: formatting

File: build/repo.props [hunk 12]
```diff
@@ -1,12 +1,20 @@
...
+    <ExcludeFromTest Include="$(RepositoryRoot)test\WebSocketClientEXE\*.csproj" />
+    <ExcludeFromTest Include="$(RepositoryRoot)test\AspNetCoreModule.Test\*.csproj" Condition="'$(OS)' != 'Windows_NT'" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <PlatformTarget>x64</PlatformTarget>
```
Comment (natemcmaster) [inline build/repo.props]: What is this for? It should probably be removed.

File: build/dependencies.props (hunk 3 above)
Comment (jkotalik) [reply build/dependencies.props]: I'll try to remove it. Seems a bit odd to have a single dll as a nupkg.

Comment (natemcmaster) [reply build/dependencies.props]: If its too hard, we can make this a cleanup item. The scope of this PR is already huge. We can add it to external dependencies in aspnet/Universe for now to unblock this.

File: README.md (hunk 6 above)
Comment (jkotalik) [reply README.md]: I'm going to drop the ANCM readme for now. @shirhatti can you update before 2.1?

File: build/dependencies.props (hunk 3 above)
Comment (jkotalik) [reply build/dependencies.props]: This is the official package: https://www.nuget.org/packages/System.Management.Automation/ Would this need to be added as an external dependency to universe?

Comment (shirhatti) [reply build/dependencies.props]: It's one of those weird packages where some random person shipped a framework assembly as a NuGet package. I believe removing the package dependency will non-trivial. I like @natemcmaster suggestion do this as a cleanup task after this PR

File: Directory.Build.props [hunk 13]
```diff
@@ -18,9 +18,4 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
-  <ItemGroup>
```
Comment (shirhatti) [inline Directory.Build.props]: Do you no longer need this for IISIntegration?

Comment (jkotalik) [reply Directory.Build.props]: It was already removed from dev, not sure why it is showing up as a diff.

File: build/dependencies.props [hunk 14]
```diff
@@ -1,43 +1,48 @@
...
-    <SystemBuffersPackageVersion>4.4.0</SystemBuffersPackageVersion>
-    <SystemIOPipelinesPackageVersion>0.1.0-alpha-002</SystemIOPipelinesPackageVersion>
-    <SystemMemoryPackageVersion>4.5.0-preview1-25902-08</SystemMemoryPackageVersion>
-    <SystemNumericsVectorsPackageVersion>4.5.0-preview1-25902-08</SystemNumericsVectorsPackageVersion>
-    <SystemRuntimeCompilerServicesUnsafePackageVersion>4.5.0-preview1-25902-08</SystemRuntimeCompilerServicesUnsafePackageVersion>
-    <SystemSecurityPrincipalWindowsPackageVersion>4.4.0</SystemSecurityPrincipalWindowsPackageVersion>
-    <SystemTextEncodingsWebUtf8PackageVersion>0.1.0-alpha-002</SystemTextEncodingsWebUtf8PackageVersion>
-    <XunitPackageVersion>2.3.0</XunitPackageVersion>
-    <XunitRunnerVisualStudioPackageVersion>2.3.0</XunitRunnerVisualStudioPackageVersion>
+    <MicrosoftWebAdministrationPackageVersion>7.0.0</MicrosoftWebAdministrationPackageVersion>
```
Comment (shirhatti) [inline build/dependencies.props]: Ughh. We should be using 10.0.0 of Microsoft.Web.Administration. That's a real .NET Standard package.
7.0.0 is another of those janky packages where someone shipped a binary from the GAC to NuGet

File: src/AspNetCore/AspNetCore.vcxproj [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>AspNetCoreModule</RootNamespace>
+    <ProjectName>AspNetCore</ProjectName>
+    <TargetName>aspnetcore</TargetName>
+    <LinkIncremental>false</LinkIncremental>
+    <WindowsTargetPlatformVersion>10.0.15063.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
+    <ConfigurationType>DynamicLibrary</ConfigurationType>
```
Comment (shirhatti) [inline src/AspNetCore/AspNetCore.vcxproj]: Nit: Is there a way to collapse these properties as opposed to set them for every configuration?

Comment (jkotalik) [reply src/AspNetCore/AspNetCore.vcxproj]: I'd assume there is, but I'd rather do this in a separate PR. I'll make an issue for it after merging.

**Comment by jkotalik** [comment]
.gitattributes are now set for .rc files, but git still believes these are binary files. I have no idea why git sees these as binary.

**Comment by natemcmaster** [comment]
Changes LGTM. :shipit:

**Comment by natemcmaster** [comment]
But of course, get the PR building on AppVeyor first, if possible

**Comment by pan-wang** [comment]
for the build warnings from IISLib, you can leave then as is. The code was duplicated from IISOOB directly. I don't want to change it for now as the code was broadly used by IIS OOB products meanwhile we don't have any unit test for it (even in IISOOB), and ANCM does not have enough test coverage for any potential change in IISLIB either. We can revisit this lib later once we have enough ANCM test coverage

**Pull request merged**